src/commands/logout.js

```diff
@@ -1,16 +1,25 @@
 import auth from '../utils/auth';
 
-Cypress.Commands.add('logout', (returnTo) => {
-  const options = {
-    clientID: Cypress.env('auth0ClientId'),
-  };
+Cypress.Commands.add('logout', (options = {}) => {
+  const { returnTo, logoutUrl } = options;
 
-  if (returnTo) {
-    options.returnTo = returnTo;
-  }
+  const builtLogoutUrl = auth.client.buildLogoutUrl({
+    clientID: Cypress.env('auth0ClientId'),
+    returnTo: returnTo,
+  });
+  cy.request(builtLogoutUrl);
 
-  const logoutUrl = auth.client.buildLogoutUrl(options);
+  /**
+   * Because we preserved cookies they might not be deleted by now. That is why
+   * we force them to be deleted.
+   *
+   * TODO: needs further investigation if this is the way to go when handling
+   * logouts in Cypress.
+   */
+  cy.clearAuth0Cookies();
 
-  cy.request('/api/auth/logout');
-  cy.reload();
+  /**
+   * default URL is the default URL that nextjs-auth0 defines and uses
+   */
+  cy.visit(logoutUrl || '/api/auth/logout');
```

Comment:
see you are using `cy.visit()` instead of `cy.request()` in this line. But above you're using `cy.request(builtLogoutUrl);`. Does it matter which method we call here?

Reply:
This is mainly because you can't visit the created `builtLogoutUrl` directly, as you get the following error in Cypress if you do:

<img width="722" alt="image" src="https://user-images.githubusercontent.com/1043668/153652433-3be5b639-554e-4bd0-b3e2-a7e03839b5ba.png">

I am not sure though if we need it or can remove it completely. It wasn't used in a previous PR (created by a nextjs-auth0 maintainer) at all: https://github.com/sir-dunxalot/cypress-nextjs-auth0/pull/14/files#. 🤔

Reply:
Ok. I think the best approach here is to use `cy.request(builtLogoutUrl)` and remove the call to `cy.visit()`. So long as we call the built logout URL somehow and test are passing then there is no need for two calls to different logout URLs.

Reply:
Resolved this conversation in favor of https://github.com/sir-dunxalot/cypress-nextjs-auth0/pull/44#discussion_r805134730.